feat: Added documenation for patternfly cli. (#4981)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/get-started/patternfly-cli.md

---
id: PatternFly CLI
title: PatternFly CLI

> section: get-started

Reviewer: the get started section is more so intended for introductory/overview information. Are you set on including these docs there? Otherwise, we could move to /developer-guides or maybe at least nest it under get-started/develop (assuming this is a dev-focused tool)?

Author: This is for both designers and developers. At the moment it's geared towards designers more.  I'm good with putting it under developer guides

---

import './get-started.css';
import { Alert } from '@patternfly/react-core';

> The [PatternFly CLI](https://github.com/patternfly/patternfly-cli) is a command-line tool for scaffolding projects, applying code changes, and running common project tasks. It is published as [`@patternfly/patternfly-cli` on npm](https://www.npmjs.com/package/@patternfly/patternfly-cli).

Reviewer: ```suggestion
The [PatternFly CLI](https://github.com/patternfly/patternfly-cli) is a command-line tool published on npm as [`@patternfly/patternfly-cli`](https://www.npmjs.com/package/@patternfly/patternfly-cli). With this tool, you can scaffold projects from built-in or custom templates, automate code updates for repetitive changes, and run workflows (including init, sync with GitHub, updates, and deploys to GitHub Pages).

The PatternFly CLI is developed in the open at [github.com/patternfly/patternfly-cli](https://github.com/patternfly/patternfly-cli), with release notes and tags published on [Releases page](https://github.com/patternfly/patternfly-cli/releases). To report issues or contribute to this tool, refer to the patternfly-cli repo for more instruction.
```

wdyt about this description, rather than the "what you can do" list? Just to build those list points into the narrative a little more 

also moved up the bit about source and releases from the end of the page

## What you can do

- **Scaffold projects** from built-in or custom templates.
- **Automate code updates** for repetitive changes.

> - **Run workflows** such as init, sync with GitHub, updates, and deploys to GitHub Pages.

Reviewer: ```suggestion
```
remove, if you're okay with folding into the intro description as I did

> ## Prerequisites

Reviewer: ```suggestion
## How do I set up the PatternFly CLI? 

### Prerequisites
```

Author: Done.

> Before you install the CLI, set up:

Reviewer: ```suggestion
Before you install the PatternFly CLI, you must set up the following:
```

1. **[Node.js and npm](https://nodejs.org/)** (supported versions **20–24**).
1. **[Corepack](https://nodejs.org/api/corepack.html)** — included with Node.js; enable it with `corepack enable` after installing Node.

> 1. **[GitHub CLI](https://cli.github.com/)** (`gh`) — required for commands that talk to GitHub.

Reviewer: ```suggestion
1. **[Node.js and npm](https://nodejs.org/)** (versions 20–24 currently supported).
1. **[Corepack](https://nodejs.org/api/corepack.html):** After Node is installed, enabled corepack via `corepack enable` on the command line.
1. **[GitHub CLI](https://cli.github.com/):** Required for commands that refer to GitHub (`gh`).
```
Is corepack enable a cli command?

> ## Install

Reviewer: ```suggestion
### Install
```

> Install the CLI globally:

Reviewer: ```suggestion
Install the PatternFly CLI globally:
```

```sh
npm install -g @patternfly/patternfly-cli
```

> ## Check your setup

Reviewer: ```suggestion
### Check your setup
```

> Run the doctor command to verify prerequisites:

Reviewer: ```suggestion
To verify prerequisites, run the `doctor` command:
```

```sh
patternfly-cli doctor
```

> To try to fix missing pieces automatically (Corepack and GitHub CLI only):

Reviewer: ```suggestion
To try to fix missing pieces automatically, add the `--fix` flag (Corepack and GitHub CLI only):
```

```sh
patternfly-cli doctor --fix
```

<Alert title="Node.js must be installed separately" variant="warning" isInline>
The `doctor --fix` option does not install or upgrade Node.js. If Node.js is missing or below version 20, install the current **LTS** release from [nodejs.org](https://nodejs.org/).

> </Alert>

Reviewer: ```suggestion
<Alert title="Node.js must be installed separately" variant="warning" isInline>
The `doctor --fix` option does not install or upgrade Node.js. If Node.js is missing or below version 20, install the current LTS release from [nodejs.org](https://nodejs.org/).
</Alert>
```

> ## Commands

Reviewer: ```suggestion
## How do I use the PatternFly CLI? 

### Commands
```

> After installation, run:

Reviewer: ```suggestion
Once the PatternFly CLI is installed, you can run the following commands via `patternfly-cli [command]`:
```

```sh
patternfly-cli [command]
```

> | Command | Description |

Reviewer: ```suggestion
| Command | Usage |
```

Author: Done.

| --- | --- |
| `doctor` | Check requirements; use `--fix` to enable Corepack and install `gh` where supported. |
| `create` | Create a new project from a template. |
| `list` | List available templates (built-in and custom). |
| `update` | Update the project to a newer PatternFly-oriented version where applicable. |
| `init` | Initialize a git repository and optionally create a GitHub repo. |
| `save` | Commit and push changes on the current branch. |
| `load` | Pull the latest changes from GitHub. |

> | `deploy` | Build and deploy the app to GitHub Pages. |

Reviewer: ```suggestion
| `doctor` | Check requirements. Add the `--fix` flag to enable Corepack and install `gh` where supported. |
| `create` | Create a new project from a template. |
| `list` | List available templates (built-in and custom). |
| `update` | Update the project to a newer PatternFly-oriented version where applicable. |
| `init` | Initialize a git repository and optionally create a GitHub repo. |
| `save` | Commit and push changes on the current branch. |
| `load` | Pull the latest changes from GitHub. |
| `deploy` | Build and deploy the app to GitHub Pages. |
```

Reviewer: "Add the `--fix` flag to enable Corepack and install `gh` where supported. |" This reads like --fix both enables Corepack and installs gh -- is that correct?

> For the most up-to-date flags and behavior, see the [PatternFly CLI README](https://github.com/patternfly/patternfly-cli/blob/main/README.md) on GitHub.

Reviewer: ```suggestion
For the most up-to-date flags and behavior guidance, refer to the [PatternFly CLI README](https://github.com/patternfly/patternfly-cli/blob/main/README.md) on GitHub.
```

> ## Custom templates

Reviewer: ```suggestion
### Custom templates
```

> You can merge your own templates with the built-ins by passing a JSON file with `--template-file` (or `-t`):

Reviewer: ```suggestion
You can merge your own templates with the built-in templates by passing a JSON file with `--template-file` (or `-t`):
```

```sh
patternfly-cli create my-app --template-file ./my-templates.json
patternfly-cli list --template-file ./my-templates.json
```

> Each entry is an object with at least `name`, `description`, and `repo` (clone URL). Optional fields include `options` (extra `git clone` arguments) and `packageManager` (`npm`, `yarn`, or `pnpm`; default is `npm`). If a custom template uses the same `name` as a built-in one, the custom definition wins.

Reviewer: ```suggestion
Each entry is an object with at least `name`, `description`, and `repo` (clone URL). Optional fields include `options` (extra `git clone` arguments) and `packageManager` (`npm` by default, or `yarn`/`pnpm`). If a custom template uses the same `name` as a built-in template, the custom definition is used.
```

Example:

```json
[
  {
    "name": "my-template",
    "description": "My custom project template",
    "repo": "https://github.com/org/repo.git",
    "options": ["--single-branch", "--branch", "main"],
    "packageManager": "npm"
  }
]
```

## Source and releases

> The CLI is developed in the open at [github.com/patternfly/patternfly-cli](https://github.com/patternfly/patternfly-cli). Report issues or contribute there; release notes and tags are published on the repository’s [Releases](https://github.com/patternfly/patternfly-cli/releases) page.

Reviewer: ```suggestion
```
moving up top?

Author: Do we even need this? I can remove it since the guidance link will take you to the readme that will have this information in it. Removing for now, if you want me to add it back let me know.
